packages/web-components/src/components/content-block/content-block.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { classMap } from 'lit-html/directives/class-map';
+import { html, property, internalProperty, LitElement, TemplateResult } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings.js';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import ifNonNull from 'carbon-web-components/es/globals/directives/if-non-null.js';
+import '../markdown/markdown';
+import styles from './content-block.scss';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * The table mapping slot name with the private property name that indicates the existence of the slot content.
+ */
+const slotExistencePropertyNames = {
+  complementary: '_hasComplementary',
+  cta: '_hasCTA',
+};
+
+/**
+ * The style scheme for the complementary content.
+ */
+export enum CONTENT_BLOCK_COMPLEMENTARY_STYLE_SCHEME {
+  /**
+   * Regular style.
+   */
+  REGULAR = 'regular',
+
+  /**
+   * A style with border.
+   */
+  WITH_BORDER = 'with-border',
+}
+
+/**
+ * Content block.
```

Comment:
Super minor thing (and also something to consider for the other ones that fall under this category) but wondering if we want to add any documentation that this and others are internal components, which is why we have not included storybook entries for them?

Reply:
Good point @jeffchew - I was thinking of marking this class as `abstract`, will try this. BTW there is no `<dds-content-block>`, because `<dds-content-block-simple>`, etc. inherits `DDSContentBlock` class instead of using `<dds-content-block>` by a composition.

Reply:
Updated, by adding `@abstract` JSDoc comment.